src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -99,68 +103,49 @@ public static void main(String[] args) throws ConfigError {
+        } catch (IOException | NullPointerException | ConfigError e) {
             LOGGER.error("Failed to create DataDictionary", e);
+            System.exit(1);
         }
 
         for (FixBean fixBean : settings.sessionsSettings) {
-            switch (fixBean.getBeginString()) {
-                case "FIX.4.0":
-                    for (File dataDict : dictionaries) {
-                        if (dataDict.getName().equals("FIX40.xml"))
-                            fixBean.setDataDictionary(dataDict.getAbsolutePath());
-                    }
-                    break;
-                case "FIX.4.1":
-                    for (File dataDict : dictionaries) {
-                        if (dataDict.getName().equals("FIX41.xml"))
-                            fixBean.setDataDictionary(dataDict.getAbsolutePath());
-                    }
-                    break;
-                case "FIX.4.2":
-                    for (File dataDict : dictionaries) {
-                        if (dataDict.getName().equals("FIX42.xml"))
-                            fixBean.setDataDictionary(dataDict.getAbsolutePath());
-                    }
-                    break;
-                case "FIX.4.3":
-                    for (File dataDict : dictionaries) {
-                        if (dataDict.getName().equals("FIX43.xml"))
-                            fixBean.setDataDictionary(dataDict.getAbsolutePath());
-                    }
-                    break;
-                case "FIX.4.4":
-                    for (File dataDict : dictionaries) {
-                        if (dataDict.getName().equals("FIX44.xml"))
-                            fixBean.setDataDictionary(dataDict.getAbsolutePath());
-                    }
-                    break;
-                case "FIX.5.0":
-                    for (File dataDict : dictionaries) {
-                        if (dataDict.getName().equals("FIX50.xml"))
-                            fixBean.setDataDictionary(dataDict.getAbsolutePath());
-                    }
-                    break;
+            for (File dataDict : dictionaries) {
+                if (dataDict.getName().equals(fixBean.getBeginString() + ".xml"))
+                    fixBean.setDataDictionary(dataDict.getAbsolutePath());
+            }
+            if (fixBean.getDataDictionary() == null || fixBean.getDataDictionary().equals("")) {
+                throw new NullPointerException("Data Dictionary does not set");
             }
         }
 
         MessageRouter<EventBatch> eventRouter = factory.getEventBatchRouter();
         MessageRouter<MessageGroupBatch> messageRouter = factory.getMessageRouterMessageGroupBatch();
         GrpcRouter grpcRouter = factory.getGrpcRouter();
 
-        run(settings, messageRouter, eventRouter, grpcRouter, resources);
+        try {
+            run(settings, messageRouter, eventRouter, grpcRouter, resources);
+        } catch (IOException e) {
+            LOGGER.error("Failed to create config file", e);
```

Comment:
I guess only creating the config file can throw such kind of error. In all another cases I catch IO exceptions.

Reply:
My point is that you that the scope in which you're assuming that this `IOException` is related only to creating a config file is too broad. Yeah, it could be true at the moment, but if `run` method will start throwing `IOException` in some other cases you'll have a misleading record in logs. You should either make log message more generic or rewrap `IOException` which was caused by the a config file in `run` method to a more narrow type and catch it here.